test/unit/deploy.js

```diff
@@ -113,6 +113,68 @@ exports['Tessel.prototype.deployScript'] = {
   },
 };
 
+exports['Tessel.prototype.restartScript'] = {
+  setUp: function(done) {
+    this.runScript = sandbox.stub(deploy, 'runScript', function() {
+      return Promise.resolve();
+    });
+    this.startPushedScript = sandbox.stub(deploy, 'startPushedScript', function() {
+      return Promise.resolve();
+    });
+
+    this.findProject = sandbox.stub(deploy, 'findProject', function(entryPoint) {
+      return Promise.resolve({
+        entryPoint: entryPoint
+      });
+    });
+
+    this.logsWarn = sandbox.stub(logs, 'warn', function() {});
+    this.logsInfo = sandbox.stub(logs, 'info', function() {});
+
+    this.tessel = TesselSimulator();
+
+    done();
+  },
+
+  tearDown: function(done) {
+    this.tessel.mockClose();
+
+    sandbox.restore();
+
+    done();
+  },
+
+  restartFromRam: function(test) {
```

Comment:
I suggest one more test that shows that an error is displayed if there is no project in the run or push directory.